src/spikeinterface/sortingcomponents/motion/dredge.py

```diff
@@ -0,0 +1,1469 @@
+        than the largest displacement possible in your recording across a span of `time_horizon_s`
+        seconds, setting this value to that number can stabilize the result and speed up
+        the algorithm (since it can do less cross-correlating).
+        By default, this is set to win-scale_um / 4, or 112.5 microns. Which can be a bit
+        large!
+    time_horizon_s : float
+        "Time horizon" parameter, in seconds. Time bins separated by more seconds than this
+        will not be cross-correlated. So, if your data has nonstationarities or changes which
+        could lead to bad cross-correlations at some timescale, it can help to input that
+        value here. If this is too small, it can make the motion estimation unstable.
+    mincorr : float, between 0 and 1
+        Correlation threshold. Pairs of frames whose maximal cross correlation value is smaller
+        than this threshold will be ignored when solving for the global displacement estimate.
+    thomas_kw, xcorr_kw, raster_kw, weights_kw
+        These dictionaries allow setting parameters for fine control over the registration
+    device : str or torch.device
+        What torch device to run on? E.g., "cpu" or "cuda" or "cuda:1".
+    """
+    @classmethod
+    def run(
+        cls,
+        recording,
+        peaks,
+        peak_locations,
+        direction,
+        rigid,
+        win_shape,
+        win_step_um,
+        win_scale_um,
+        win_margin_um,
+        verbose,
+        progress_bar,
+        extra,
+        **method_kwargs,
+    ):
+
+
+        outs = dredge_ap(
+            recording,
+            peaks,
+            peak_locations,
+            direction=direction,
+            rigid=rigid,
+            win_shape=win_shape,
+            win_step_um=win_step_um,
+            win_scale_um=win_scale_um,
+            win_margin_um=win_margin_um,
+            extra_outputs=(extra is not None),
+            progress_bar=progress_bar,
+            **method_kwargs,
+        )
+
+        if extra is not None:
+            motion, extra_ = outs
+            extra.update(extra_)
+        else:
+            motion = outs
+        return motion
+
+
+# @TODO : Charlie I started very small refactoring, I let you continue
+def dredge_ap(
+    recording,
+    peaks,
+    peak_locations,
+    direction="y",
+    rigid=False,
+    # nonrigid window construction arguments
+    win_shape="gaussian",
+    win_step_um=400,
+    win_scale_um=450,
+    win_margin_um=None,
+    bin_um=1.0,
+    bin_s=1.0,
+    max_disp_um=None,
+    time_horizon_s=1000.,
+    mincorr=0.1,
+    # weights arguments
+    do_window_weights=True,
+    weights_threshold_low=0.2,
+    weights_threshold_high=0.2,
+    mincorr_percentile=None,
+    mincorr_percentile_nneighbs=None,
+    # raster arguments
+    amp_scale_fn=None,  ## @Charlie this one is not used anymore
+    post_transform=np.log1p,  ###@this one is directly transimited to weight_correlation_matrix() and so get_wieiith()
```

Comment:
actually we can probably not even transmit it. we can apply the function to the threshold in advance before passing it to weight_correlation_matrix to reduce the number of params. I can do this in my PR.

Reply:
ok. I left the code  like this for the moment.